incubator/tools-typescript/src/platforms.ts

```diff
@@ -0,0 +1,302 @@
+  visionos: "@callstack/react-native-visionos",
+};
+
+/**
+ * @returns a PlatformInfo with a package name and the module suffixes set
+ */
+function createPlatformInfo(platform: string, npmName?: string): PlatformInfo {
+  const pkgName = npmName || fallbackPackages[platform];
+  if (!pkgName) {
+    throw new Error(
+      `Unable to determine package name for platform ${platform}`
+    );
+  }
+  return {
+    name: platform,
+    pkgName: npmName || fallbackPackages[platform],
+    suffixes: platformExtensions(platform)
+      .concat("")
+      .map((s) => `.${s}`),
+  };
+}
+
+/**
+ * This determines supported react native platforms for a package given the bundle config if it is marked as an app
+ * @returns an array of react-native platforms that the package supports
+ */
+export function platformsFromKitConfig(
+  kitConfig: KitConfig | undefined
+): string[] | undefined {
+  const found = new Set<string>();
+  if (kitConfig?.kitType === "app") {
+    // for packages marked as an 'app', determine available platforms based on the bundle configuration
+    for (const bundleConfig of coerceArray(kitConfig.bundle)) {
+      const { platforms, targets } = bundleConfig || {};
+      if (platforms) {
+        for (const platform in platforms) {
+          found.add(platform);
+        }
+      }
+      for (const target of coerceArray(targets)) {
+        found.add(target);
+      }
+    }
+  }
+  return found.size > 0 ? Array.from(found) : undefined;
+}
+
+/**
+ * Load platform info for available platforms for a given package
+ *
+ * @param pkgRoot the root directory of the package
+ * @param manifest the loaded package manifest
+ * @param platformOverride override platform detection with a specific set of platforms
+ * @returns a map of platform name to PlatformInfo
+ */
+export function loadPackagePlatformInfo(
+  pkgRoot: string,
+  manifest: PackageManifest,
+  platformOverride?: string[]
+): Record<string, PlatformInfo> {
+  // load the available platforms for the package from the react-native config files (and dependencies)
+  const available = getAvailablePlatforms(pkgRoot);
+
+  // the platforms are set specifically, found in the kit config, or equal all available platforms
+  const platforms =
+    platformOverride ||
+    platformsFromKitConfig(
+      getKitConfigFromPackageManifest(manifest, pkgRoot)
+    ) ||
+    Object.keys(available);
+  const result: Record<string, PlatformInfo> = {};
+
+  for (const platform of platforms) {
+    result[platform] = createPlatformInfo(platform, available[platform]);
+  }
+  return result;
+}
+
+export function parseFileReference(
+  file: string,
+  extensions: Set<string>,
+  ignoreSuffix?: boolean
+): ParsedFileReference {
+  let ext = path.extname(file);
+  let suffix = "";
+  let base = file;
+
+  if (extensions.has(ext)) {
+    if (ext === ".ts" || ext === ".tsx") {
+      const dtsExt = ".d" + ext;
+      if (extensions.has(dtsExt) && file.endsWith(dtsExt)) {
```

Comment:
Can't we assume `extensions.has(dtsExt)` is always true here?
```suggestion
      if (file.endsWith(dtsExt)) {
```
And on that note, do we even need `.d.tsx` in the set?

Reply:
I don't want to make that assumption because it eventually depends on the parsing set. In some cases it may be configuration driven. I am removing .d.tsx, I don't think they are a thing.